dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/PropertyService.java

```diff
@@ -152,9 +153,131 @@ public int hashCode() {
         Map.Entry entry : map.entrySet())
             if (entry.getKey() instanceof String && PropertyService.isPassword((String) entry.getKey()))
                 entry.setValue("******");
-
+            else if(entry.getKey() instanceof String && entry.getValue() instanceof String && ((String) entry.getKey()).toLowerCase().contains("url"))
+                entry.setValue(hidePasswordsInURL((String) entry.getValue()));
         return map;
     }
+    
+    public static String hidePasswordsInURL(String url) {
+        int first = url.indexOf(":");
+        int second = url.indexOf(":", first + 1);
+        int third = url.indexOf(":", second + 1);
+        int progress = -1;
+        StringBuilder sb = new StringBuilder();
+        String lowerCaseURL = url.toLowerCase();
+        
+        Pattern alphanumericPattern = Pattern.compile("[\\w]*");
```

Comment:
Could be more efficient to compile the pattern once instead of every time through the method, but that would require compiling it once on a non-trace path, so might not be worth it

Reply:
Yeah, not sure about the performance trade off there, but I'm probably more concerned about optimizing the non-trace path.